components/markdown/markdown.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import messageHtmlToComponent from 'utils/message_html_to_component';
+import EmojiMap from 'utils/emoji_map';
+import * as TextFormatting from 'utils/text_formatting';
+
+type Options = {
+    mentionHighlight?: boolean;
+}
+
+type Props = {
+
+    /*
+     * An object mapping channel names to channels for the current team
+     */
+    channelNamesMap?: object;
+
+    /*
+     * An array of URL schemes that should be turned into links. Anything that looks
+     * like a link will be turned into a link if this is not provided.
+     */
+    autolinkedUrlSchemes?: Array<any>;
```

Comment:
The URL schemes are strings like `'https'` and `'mailto'`, so this can be
```suggestion
    autolinkedUrlSchemes?: Array<string>;
```